src/compiler/types.ts

```diff
@@ -2365,6 +2365,11 @@ namespace ts {
         sourceFiles: SourceFile[];
     }
 
+    export interface JsonSourceFile extends SourceFile {

```

Comment:
This shouldn't extend from `SourceFile`. There's an existing `SourceFileLike` interface, could you use that?